packages/components/src/test/instance-filter-builder/Utils.test.ts

```diff
@@ -165,42 +165,67 @@ describe("filterRuleValidator", () => {
         value: {
           valueFormat: PropertyValueFormat.Primitive,
           value: undefined,
-          displayValue: "Invalid",
+          displayValue: "[{}]",
         },
       }),
     ).to.be.eq("instance-filter-builder.error-messages.not-a-number");
   });
 
-  it("returns error message for invalid quantity rule", () => {
+  it(`returns error message for invalid array elements`, () => {
     expect(
       filterRuleValidator({
         id: "test-id",
         groupId: "test-group-id",
-        property: quantityProperty,
+        property: numericProperty,
         operator: PropertyFilterRuleOperator.IsEqual,
         value: {
           valueFormat: PropertyValueFormat.Primitive,
-          value: undefined,
-          displayValue: "Invalid",
+          value: "[[10], [{}]]",
+          displayValue: "[[10], [{}]]",
         },
       }),
-    ).to.be.eq("instance-filter-builder.error-messages.invalid");
+    ).to.be.eq("instance-filter-builder.error-messages.not-a-number");
   });
 
-  it("does not return error message for valid numeric rule", () => {
+  it("returns error message for invalid quantity rule", () => {
     expect(
       filterRuleValidator({
         id: "test-id",
         groupId: "test-group-id",
-        property: numericProperty,
+        property: quantityProperty,
         operator: PropertyFilterRuleOperator.IsEqual,
         value: {
           valueFormat: PropertyValueFormat.Primitive,
-          value: 10,
-          displayValue: "10",
+          value: undefined,
+          displayValue: "Invalid",
         },
       }),
-    ).to.be.undefined;
+    ).to.be.eq("instance-filter-builder.error-messages.invalid");
+  });
+
+  [
+    { val: 10, testCase: "number" },
+    { val: [10], testCase: "number array" },
+    { val: [[10]], testCase: "nested number array" },
+    { val: "[10]", testCase: "number array as string" },
+    { val: "[[10]]", testCase: "nested number array as string" },
+    { val: "[[10], [10]]", testCase: "nested number arrays as string" },
```

Comment:
Could the test names be more specific about what `PropertyValue` type they test? E.g. I'm not sure which `PropertyValue` type case is covered with the `number array as string` test.

Reply:
Not really sure if I understood this correctly, please check the latest commit and let me know if it still does not look right.

Reply:
The change did not solve my concern.

`filterRuleValidator` validates a `PropertyFilterBuilderRule`, which contains a `PropertyValue`. Here we're testing if the function correctly handles different types of its subtype - `PrimitiveValue`. The `value` part of `PrimitiveValue`, which varies in this test, is defined as `Primitives.Value`, which is defined as `Text | String | ShortDate | Boolean | Numeric | Enum | Point | Composite | InstanceKey`. I think the test cases should be based on those sub-types. 

For example, I'm not sure which of those sub-types is represented by `{ val: [10], testCase: "number array" }` test case. Maybe `Point`? In that case I'd like the test to be named "Point" or something like that.

Reply:
Yeah, now I understand what you meant about `PropertyValue` types. Initially, I thought you were talking about the types  `PrimitiveValue | StructValue | ArrayValue`. 

But now that we've simplified tests and implementation, I don't think these extra test cases are needed.